RecoMuon/TrackerSeedGenerator/plugins/MuonHLTSeedMVAClassifier.cc

```diff
@@ -0,0 +1,282 @@
+
+  const l1t::MuonBxCollection& l1Muons = iEvent.get(l1MuonToken_);
+  const reco::RecoChargedCandidateCollection& l2Muons = iEvent.get(l2MuonToken_);
+  const TrajectorySeedCollection& seeds = iEvent.get(seedToken_);
+
+  std::vector<std::pair<unsigned, double>> pairSeedIdxMvaScoreB = {};
+  std::vector<std::pair<unsigned, double>> pairSeedIdxMvaScoreE = {};
+  for (auto& seed : seeds) {
+    const GlobalVector global_p =
+        trkGeom->idToDet(seed.startingState().detId())->surface().toGlobal(seed.startingState().parameters().momentum());
+
+    bool isB = (std::abs(global_p.eta()) < etaEdge_);
+
+    if (doSort_) {
+      if (isB) {
+        if (nSeedsMaxB_ <= 0) {
+          continue;
+        }
+      } else {
+        if (nSeedsMaxE_ <= 0) {
+          continue;
+        }
+      }
+    } else {
+      if (isB) {
+        if (mvaCutB_ > 1.0) {
+          continue;
+        } else if (mvaCutB_ <= 0.) {
+          result->emplace_back(seed);
+          continue;
+        }
+      } else {
+        if (mvaCutE_ > 1.0) {
+          continue;
+        } else if (mvaCutE_ <= 0.) {
+          result->emplace_back(seed);
+          continue;
+        }
+      }
+    }
+
+    double mva = getSeedMva(mvaEstimator_, seed, global_p, l1Muons, l2Muons);
+
+    double score = 1. / (1. + std::exp(-1. * mva));
+    bool passMva = isB ? score > mvaCutB_ : score > mvaCutE_;
+    if (!passMva)
+      continue;
+
+    if (doSort_) {
+      if (isB)
+        pairSeedIdxMvaScoreB.push_back(std::make_pair(&seed - &seeds.at(0), score));
+      else
+        pairSeedIdxMvaScoreE.push_back(std::make_pair(&seed - &seeds.at(0), score));
+    } else {
+      result->emplace_back(seed);
+    }
+  }
+
+  if (doSort_) {
+    std::sort(pairSeedIdxMvaScoreB.begin(), pairSeedIdxMvaScoreB.end(), sortByMvaScore);
+    std::sort(pairSeedIdxMvaScoreE.begin(), pairSeedIdxMvaScoreE.end(), sortByMvaScore);
+
+    for (auto i = 0U; i < pairSeedIdxMvaScoreB.size(); ++i) {
+      if ((int)i == nSeedsMaxB_)
+        break;
+      const auto& seed(seeds.at(pairSeedIdxMvaScoreB.at(i).first));
+      result->emplace_back(seed);
+    }
+
+    for (auto i = 0U; i < pairSeedIdxMvaScoreE.size(); ++i) {
+      if ((int)i == nSeedsMaxE_)
+        break;
+      const auto& seed(seeds.at(pairSeedIdxMvaScoreE.at(i).first));
+      result->emplace_back(seed);
+    }
+  }
+
+  iEvent.put(std::move(result));
+}
+
+double MuonHLTSeedMVAClassifier::getSeedMva(const pairSeedMvaEstimator& pairMvaEstimator,
+                                            const TrajectorySeed& seed,
+                                            const GlobalVector& global_p,
+                                            const l1t::MuonBxCollection& l1Muons,
+                                            const reco::RecoChargedCandidateCollection& l2Muons) {
+  double mva = 0.;
+  if (std::abs(global_p.eta()) < etaEdge_) {
+    mva = pairMvaEstimator.first->computeMva(seed, global_p, l1Muons, minL1Qual_, l2Muons, isFromL1_);
```

Comment:
I see the `isFromL1` is passed to `SeedMvaEstimator` here. As the `isFromL1_` is set in the constructor, I think it would make `SeedMvaEstimator` easier to reason if the `isFromL1_` would be given in `SeedMvaEstimator` constructor. Then it would be clear that the `isFromL1` should correspond the model and can not change between calls to `computeMva`.